packages/core/src/browser/frontend-application-module.ts

```diff
@@ -340,4 +341,9 @@ export const frontendApplicationModule = new ContainerModule((bind, unbind, isBo
 
     bind(AuthenticationService).to(AuthenticationServiceImpl).inSingletonScope();
     bind(DecorationsService).to(DecorationsServiceImpl).inSingletonScope();
+
+    bind(WindowContribution).toSelf().inSingletonScope();
+    for (const contribution of [CommandContribution, KeybindingContribution, MenuContribution]) {
+        bind(contribution).toService(WindowContribution);
+    }
```

Comment:
You can use the utility `bindViewContribution` to conveniently perform these 4 lines.

Reply:
Hi colin, I don't think that I can use the bindViewContribution utility because WindowContribution  doesn't contribute to a widget. It's only used to support the 'New Window' command for both electron and browser targets and doesn't add an user interface or a view.

Reply:
I don't believe anything in the `bindViewContribution` logic (other than the name 😊) expects you to contribute a `Widget`.
```typescript
export function bindViewContribution<T extends AbstractViewContribution<any>>(bind: interfaces.Bind, identifier: interfaces.Newable<T>): interfaces.BindingWhenOnSyntax<T> {
    const syntax = bind<T>(identifier).toSelf().inSingletonScope();
    bind(CommandContribution).toService(identifier);
    bind(KeybindingContribution).toService(identifier);
    bind(MenuContribution).toService(identifier);
    return syntax;
}
```
It just binds to self, then the most common contribution points - happily, the same three you need.

Reply:
Fair enough, that generic gets in the way, even though it isn't required for the code to work...
```typescript
export function bindViewContribution<T extends AbstractViewContribution<any>>(bind: interfaces.Bind, identifier: interfaces.Newable<T>): interfaces.BindingWhenOnSyntax<T> {
    const syntax = bind<T>(identifier).toSelf().inSingletonScope();
    bind(CommandContribution).toService(identifier);
    bind(KeybindingContribution).toService(identifier);
    bind(MenuContribution).toService(identifier);
    return syntax;
}
```